xload: Missing key causes key collision check to report false positive

### Issue

When a key is missing that has a prefix as well, in that case, collision detection reports a false positive.
The latter example shows how missing key `SERVER_ADDRESS` causes the error output.

<details><summary>Working example</summary>

```go
package main

import (
	"context"
	"fmt"
	"time"

	"github.com/gojekfarm/xtools/xload"
	xloadtype "github.com/gojekfarm/xtools/xload/type"
)

type Pprof struct {
	Enabled bool               `env:"ENABLED"`
	Address xloadtype.Listener `env:"ADDRESS"`
}

type ServerConfig struct {
	HTTP *xloadtype.Listener `env:"ADDRESS"`
}

type Config struct {
	Pprof         Pprof         `env:",prefix=PPROF_"`
	Server        ServerConfig  `env:",prefix=SERVER_"`
	ShutdownDelay time.Duration `env:"SHUTDOWN_DELAY"`
}

func main() {
	ml := xload.MapLoader{
		"LOG_ENABLED":    "true",
		"LOG_LEVEL":      "debug",
		"PPROF_ENABLED":  "true",
		"PPROF_ADDRESS":  ":9090",
		"SHUTDOWN_DELAY": "4s",
		"SERVER_ADDRESS": ":8000",
	}

	cfg := new(Config)
	if err := xload.Load(context.TODO(),
		cfg,
		xload.SerialLoader(ml, xload.OSLoader()),
	); err != nil {
		panic(err)
	}

	fmt.Printf("%+v\n", cfg)
}
```

</details> 

<details><summary>Problematic Example</summary>

```go
package main

import (
	"context"
	"fmt"
	"time"

	"github.com/gojekfarm/xtools/xload"
	xloadtype "github.com/gojekfarm/xtools/xload/type"
)

type Pprof struct {
	Enabled bool               `env:"ENABLED"`
	Address xloadtype.Listener `env:"ADDRESS"`
}

type ServerConfig struct {
	HTTP *xloadtype.Listener `env:"ADDRESS"`
}

type Config struct {
	Pprof         Pprof         `env:",prefix=PPROF_"`
	Server        ServerConfig  `env:",prefix=SERVER_"`
	ShutdownDelay time.Duration `env:"SHUTDOWN_DELAY"`
}

func main() {
	ml := xload.MapLoader{
		"LOG_ENABLED":    "true",
		"LOG_LEVEL":      "debug",
		"PPROF_ENABLED":  "true",
		"PPROF_ADDRESS":  ":9090",
		"SHUTDOWN_DELAY": "4s",
		//"SERVER_ADDRESS": ":8000",
	}

	cfg := new(Config)
	if err := xload.Load(context.TODO(),
		cfg,
		xload.SerialLoader(ml, xload.OSLoader()),
	); err != nil {
		panic(err)
	}

	fmt.Printf("%+v\n", cfg)
}
```

</details> 

```shell
panic: xload: key collisions detected for keys: [SERVER_]
```

--- a/xload/load.go
+++ b/xload/load.go
@@ -148,6 +148,10 @@ func doProcess(ctx context.Context, obj any, tagKey string, loader Loader) error
 					return &ErrRequired{key: meta.key}
 				}
 
+				if val == "" && isNilStructPtr {
+					continue
+				}
+
 				if ok, err := decode(fVal, val); ok {
 					if err != nil {
 						return &ErrDecode{

--- a/xload/load_struct_test.go
+++ b/xload/load_struct_test.go
@@ -4,12 +4,15 @@ import (
 	"context"
 	"encoding/json"
 	"errors"
+	"net"
 	"reflect"
 	"testing"
 	"time"
 
 	"github.com/gotidy/ptr"
 	"github.com/stretchr/testify/assert"
+
+	xloadtype "github.com/gojekfarm/xtools/xload/type"
 )
 
 type House struct {
@@ -398,4 +401,25 @@ func TestLoad_JSON(t *testing.T) {
 	}
 
 	runTestcases(t, testcases)
+}
+
+func TestLoad_keyCollisions(t *testing.T) {
+	t.Run("MissingNestedNillableKeyRegression", func(t *testing.T) {
+		type ServerConfig struct {
+			HTTP *xloadtype.Listener `env:"ADDRESS"`
+		}
+		type Config struct {
+			Server ServerConfig `env:",prefix=SERVER_"`
+		}
+
+		cfg1 := new(Config)
+		assert.NoError(t, Load(context.TODO(), cfg1))
+		assert.Nil(t, cfg1.Server.HTTP)
+
+		cfg2 := new(Config)
+		assert.NoError(t, Load(context.TODO(), cfg2, MapLoader{"SERVER_ADDRESS": "127.0.0.1:80"}))
+		assert.NotNil(t, cfg2.Server.HTTP)
+		assert.Equal(t, net.IPv4(127, 0, 0, 1), cfg2.Server.HTTP.IP)
+		assert.Equal(t, 80, cfg2.Server.HTTP.Port)
+	})
 }

